internal/pkg/cli/svc_deploy.go

```diff
@@ -364,13 +366,40 @@ func workloadManifest(in *workloadManifestInput) (interface{}, error) {
 	if err != nil {
 		return nil, fmt.Errorf("apply environment %s override: %s", in.envName, err)
 	}
-
 	if err := envMft.Validate(); err != nil {
 		return nil, fmt.Errorf("validate manifest against environment %s: %s", in.envName, err)
 	}
 	return envMft, nil
 }
 
+func isManifestCompatibleWithEnvironment(mft manifest.WorkloadManifest, envName string, env versionFeatureGetter) error {
+	availableFeatures, err := env.AvailableFeatures()
+	if err != nil {
+		return fmt.Errorf("get available features of the environment %s stack: %w", envName, err)
+	}
+	exists := struct{}{}
+	available := make(map[string]struct{})
```

Comment:
cool!

Reply:
Why do we use `struct{}` instead of `bool`?

Reply:
Because struct{} take 0 byte

Reply:
After some investigation i think I am onboard now! But not for the memory reason which I think the benefit is so small but for the readability reason: it helps to avoid the "tri-state boolean" brought by `map[string]bool`.